arviz/data/inference_data.py

```diff
@@ -173,6 +175,53 @@ def __delattr__(self, group):
     def _groups_all(self):
         return self._groups + self._groups_warmup
 
+    def extend_xr_method(func):  # pylint: disable=no-self-argument
+        """Wrapper to add methods to InferenceData Class"""
+
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+
+            _filter = kwargs.pop("filter_groups", None)
+            _groups = kwargs.pop("groups", None)
+            _inplace = kwargs.pop("inplace", False)
+            self = args[0]
+            args = list(args)
+
+            out = self if _inplace else deepcopy(self)
+
+            groups = self._group_names(_groups, _filter)
+            for group in groups:
+                xr_data = getattr(out, group)
+                args[0] = xr_data
+                xr_data = func(*args, **kwargs)
+                setattr(out, group, xr_data)
+
+            return None if _inplace else out
+
+        description = """\
+        This method is extended from xarray.Dataset methods. For more info see :meth:`xarray:xarray.Dataset.{method_name}`
+        """.format(
+            method_name=func.__name__
+        )
+        params = """\
+            Parameters
+            ----------
+            groups: str or list of str, optional
+                Groups where the selection is to be applied. Can either be group names
+                or metagroup names.
+            filter_groups: {None, "like", "regex"}, optional, default=None
+                If `None` (default), interpret groups as the real group or metagroup names.
+                If "like", interpret groups as substrings of the real group or metagroup names.
+                If "regex", interpret groups as regular expressions on the real group or
+                metagroup names. A la `pandas.filter`.
+            inplace: bool, optional
+                If ``True``, modify the InferenceData object inplace,
+                otherwise, return the modified copy.
+            """
+        wrapped.__doc__ = description + params
```

Comment:
Is there any whitespace left between desc and params?

Reply:
No, I will correct it.